Interactivity for neverhaveiever buttons

DanPlayz0 opened the pull request:
This PR is meant to add interactivity to the neverhaveiever buttons. This is was a fun little side-project.

Here's a video of it working. Suggest any edits, if there's something you'd like.

https://github.com/user-attachments/assets/f6e87b88-d424-4062-a593-f72454353cdc

mezotv reviewed (changes requested):
Could you remove the package-lock file and get rid of anything that isnt part of the focker integration?

GopalVerma1303 commented:
Sure

mezotv commented:
I noticed you have the vps/vds workflow unchecked is this something you are wanting to add to this aswell?

GopalVerma1303 commented:
> I noticed you have the vps/vds workflow unchecked is this something you are wanting to add to this aswell?

Absolutely, I'd love to incorporate the VPS/VDS workflow. Could we create a new issue to handle this separately and keep our current focus intact? Thanks!

mezotv commented:
> > I noticed you have the vps/vds workflow unchecked is this something you are wanting to add to this aswell?
> 
> Absolutely, I'd love to incorporate the VPS/VDS workflow. Could we create a new issue to handle this separately and keep our current focus intact? Thanks!
> 

Sure just create an issue and I will assign it to you!

GopalVerma1303 commented:
I accidentally closed the merge request. Could you please re-review and merge it into the main branch? Thanks!

mezotv commented:
> I accidentally closed the merge request. Could you please re-review and merge it into the main branch? Thanks!

Let me test it together with the auto deploy workflow then I will!

GopalVerma1303 commented:
Could you please specify the hosting provider and share instructions on how to obtain the required credentials for the deployment? Thanks!

mezotv commented:
For the workflow this might be a good Inspiration https://github.com/EddieHubCommunity/BioDrop/blob/main/.github/workflows/publish-ghcr.yml

mezotv commented:
> Hey! I've created Issue #46 but didn't have permission to add labels. Could you please add the "hacktoberfest" label? Thanks!

Alright there you go!

mezotv commented on Dockerfile:
```diff
@@ -0,0 +1,83 @@
+# set hostname to localhost
+ENV HOSTNAME "0.0.0.0"
+
+CMD ["node", "server.js"]
```
What is that supposed to do? Server.js isn't a file we have.

GopalVerma1303 replied:
The server.js file in `.next/standalone/server.js` in Next.js is a minimal server that can be used to start your Next.js application in production. It is generated when you build your application with the `output: standalone` option in your next.config.js file.

mezotv replied:
Id much rather not have this there as well as the standalone extension. Could you get rid of those?

GopalVerma1303 replied:
I have made some updates. Kindly review : [e78340f](https://github.com/Would-You-Bot/website/pull/41/commits/e78340f08c9b5e39663de7230647a70eb9533ea2)

mezotv reviewed (changes requested):
In the code I saw port 3000 being used a lot is it possible to switch that since our production server uses port 2123 for the website.

GopalVerma1303 commented:
You can use this command run your container: `docker run -p 2123:2123 would-you-bot-website`

mezotv commented:
> You can use this command to run your container: `docker run -p 2123:2123 would-you-bot-website`

Could you revert the changes you have done to the sitemap?

mezotv commented on next.config.js:
```diff
@@ -28,6 +28,8 @@ const nextConfig = {
     ];
   },
+
+  output: 'standalone',
```
Remove this please!

GopalVerma1303 replied:
Sure

mezotv commented on public/sitemap-0.xml:
```diff
@@ -1,15 +1,15 @@
-<url><loc>https://wouldyoubot.gg/reddit</loc><lastmod>2023-09-20T18:55:10.103Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
-<url><loc>https://wouldyoubot.gg/support</loc><lastmod>2023-09-20T18:55:10.103Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
-<url><loc>https://wouldyoubot.gg/terms</loc><lastmod>2023-09-20T18:55:10.103Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
-<url><loc>https://wouldyoubot.gg/vote</loc><lastmod>2023-09-20T18:55:10.103Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
-<url><loc>https://wouldyoubot.gg/blog/welcome</loc><lastmod>2023-09-20T18:55:10.103Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/blog</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/commands</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/invite</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/legal</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/legal-de</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/privacy</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/reddit</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/support</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/terms</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/vote</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/blog/welcome</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
 </urlset>
```
Revert the changes here!

mezotv commented on README.md:
```diff
@@ -6,10 +6,12 @@ The official repository for the [Would You website](https://wouldyoubot.gg)
-- [Development](#development)
-- [About Bot](#about-Bot)
-- [Contributors](#contributors)
+- [Would You Web](#would-you-web)
+  - [Table of Contents](#table-of-contents)
+  - [Introduction](#introduction)
+  - [Development](#development)
+  - [About Bot](#about-bot)
+  - [Contributors](#contributors)
 
```
Revert those changes please!

GopalVerma1303 replied:
Sure

mezotv commented:
Will be checking this later thank you!

mezotv commented:
Alright, I checked it and it seems to work well, tho I have a question how would the github workflow for this work?

mezotv commented:
- Get rid of the redundant if statements for different lockfile types (npm, yarn, pnpm) since only one lockfile should be pushed to GitHub.

- Ensure that the code installs the necessary package manager (pnpm) before installing dependencies or building the app to avoid dependency conflicts.

- Consider changing file permissions after the build process to enhance security.

- Explicitly expose the port used by the application to ensure proper networking.

mezotv commented:
Hey @GopalVerma1303, will the workflow be added to this pr as well?

GopalVerma1303 commented:
> Hey @GopalVerma1303, will the workflow be added to this pr as well?

Hey! I am planning to creating a new pr for this(Issue: #46) so as to keep our current focus intact.

mezotv commented:
> > Hey @GopalVerma1303, will the workflow be added to this pr as well?
> 
> Hey! I am planning to creating a new pr for this(Issue: #46) so as to keep our current focus intact.

Hmm sounds good give me around an hour and I will test this!

mezotv reviewed (approved):
I got a bit confused with the whole server setup but I pushed it to the server and it's working well. Thanks for this pull request!

DanPlayz0 committed:
fix(typo): Parity with the discord client

DanPlayz0 committed:
feat(neverhaveiever): interactivity to the buttons under the embed

DanPlayz0 committed:
fix(neverhaveiever): Dismiss vote message type

dpaulos6 reviewed (changes requested):
If you change the theme while a custom message is visible, the message will be on top of the original message. The feature is a bit unreliable and needs improvement.


https://github.com/user-attachments/assets/c4bffca3-5d24-418d-8220-7cba1497dd3b

fb-sean commented on components/Embeds/NeverHaveIEverEmbed.tsx:
```diff
@@ -107,7 +120,93 @@ const NeverHaveIEverEmbed: FC<MainProps> = ({ replayedRounds }) => {
+            slot="embeds"
+            color={haveDone ? "#0091ff" : "#f00404"}
+            image={haveDone == null 
+              ? "https://quickchart.io/chart?c=%7Btype%3A%27outlabeledPie%27%2Cdata%3A%7Blabels%3A%5B%27Have%27%2C%27Have+not%27%5D%2Cdatasets%3A%5B%7BbackgroundColor%3A%5B%27%230091ff%27%2C%27%23f00404%27%5D%2Cdata%3A%5B1%2C1%5D%7D%5D%7D%2Coptions%3A%7Bplugins%3A%7Blegend%3Afalse%2Coutlabels%3A%7Btext%3A%27%25l+%25p%27%2Ccolor%3A%27white%27%2Cstretch%3A35%2Cfont%3A%7Bresizable%3Atrue%2CminSize%3 ...
```
I would actually copy the image instead of embed it by link to avoid third party requests from the user

dpaulos6 replied:
I agree, it would probably load instantly and would be easier to handle.

fb-sean replied:
Didn't even think about that, only said that bcs of user privacy xD

DanPlayz0 replied:
Implemented in [5b0bd6d](https://github.com/Would-You-Bot/website-app/pull/41/commits/5b0bd6db0ce87754a3c78cfd332297f286fd1032)

DanPlayz0 committed:
Prevent custom message from going above

DanPlayz0 commented:
> If you change the theme while a custom message is visible, the message will be on top of the original message. The feature is a bit unreliable and needs improvement.
> 
>  Recording.2024-07-30.123116.mp4

I've fixed this with the latest commit. Although not the best fix for it, it does now work...

DanPlayz0 committed:
Local files (privacy & faster)

mezotv commented:
> I've fixed this with the latest commit. Although not the best fix for it, it does now work...

Could you use the webp format for the images instead of png?

dpaulos6 committed:
Improvements

On theme switch, now maintains the message and doesnt change order.
Also changed image file format to `.webp` as requested by @mezotv .

dpaulos6 commented:
My latest commit ([63ef884](https://github.com/Would-You-Bot/website-app/pull/41/commits/63ef88452ecb0f02b34d25fea55ec42e3a7bff2b)) addresses the file format change requested by @mezotv. Also improves the custom message behavior on theme switch, as it was currently removing the message after switching themes and now remains there in the same order.

dpaulos6 committed:
implementing ephemeral dismiss click

mezotv commented:
@dpaulos6 Would you mind attaching a video for me to check since I am currently on vacation?

dpaulos6 committed:
fixing some buttons and color mode

DanPlayz0 commented:
Related to [10c91ae](https://github.com/Would-You-Bot/website-app/pull/41/commits/10c91ae43edd7d4a48b8548cbfc93a288238ee98), might as well delete my `EphemeralRow` component...

dpaulos6 commented:
> Related to [10c91ae](https://github.com/Would-You-Bot/website-app/pull/41/commits/10c91ae43edd7d4a48b8548cbfc93a288238ee98), might as well delete my `EphemeralRow` component...

Oh yea, completely forgot.

dpaulos6 committed:
deleting custom ephemeral component

DanPlayz0 committed:
Added myself to staffs.json

dpaulos6 committed:
adding dan to contributors

dpaulos6 commented:
As soon as I know the fix to the discord buttons without manually adding flex, this PR will be good to go afaik.
Anything missing or any cool stuff that could be added, just add a comment here.

DanPlayz0 commented:
Should the `New Question` button also be made interactive?

mezotv commented:
> Should the `New Question` button also be made interactive?

If possible yes while also clearing the voted states

dpaulos6 committed:
Fixing attribute incorrect usage

DanPlayz0 committed:
added interactivity to `New Question` button.

DanPlayz0 committed:
Whoops! console.log was left, now removed.

DanPlayz0 commented:
Here's the current version of these commits. Although there seems to be a bug with light theme and the ephemeral message text color and a random curly bracket where it says "Click to see command`}`".

https://github.com/user-attachments/assets/9ca4d041-3f2d-41dc-b9e0-d6fc421d8b2d

dpaulos6 committed:
correcting theme color issue

fb-sean commented:
Can someone somehow remove me from this PR I don't want to receive notifies for every commit change 😭

//Edit: Found the unsubscribe button...

dpaulos6 commented:
> Here's the current version of these commits. Although there seems to be a bug with light theme and the ephemeral message text color and a random curly bracket where it says "Click to see command`}`".
> 
>  2024-08-02.00-53-31.mp4

I just fixed the color bug on the light theme, for some reason some components are not applying white theme.

Related to the `}` char at the end of the message, it's a very weird bug and it's not present in any part of the code. Can you read the docs on [skyra/discord-components - migrating to v4](https://github.com/skyra-project/discord-components/blob/main/packages/core/UPGRADING-TO-V4.md) and see if you can find anything that needs changing? I recently updated the version of it to use the ephemeral hook, so can you confirm for me everything is correct with the docs on v4?

DanPlayz0 commented:
> > Here's the current version of these commits. Although there seems to be a bug with light theme and the ephemeral message text color and a random curly bracket where it says "Click to see command`}`".
> > 2024-08-02.00-53-31.mp4
> 
> I just fixed the color bug on the light theme, for some reason some components are not applying white theme.
> 
> Related to the `}` char at the end of the message, it's a very weird bug and it's not present in any part of the code. Can you read the docs on [skyra/discord-components - migrating to v4](https://github.com/skyra-project/discord-components/blob/main/packages/core/UPGRADING-TO-V4.md?rgh-link-date=2024-08-02T13%3A29%3A17Z) and see if you can find anything that needs changing? I recently updated the version of it to use the ephemeral hook, so can you confirm for me everything is correct with the docs on v4?

The rouge `}` at the end of the message, after enabling `command={true}`, is now more obvious that is within the source code of the package. I have submitted a PR (https://github.com/skyra-project/discord-components/pull/437) over there to fix this. Might take a bit to get it approved.

Here's the another screenshot (with the command set to true) that you can see the curly brace,
![image](https://github.com/user-attachments/assets/21e3d19e-c2cc-4d2c-bccf-0a29f2e7b448)

DanPlayz0 committed:
Set `command` to true for discord-reply

DanPlayz0 committed:
light theme variables

mezotv commented:
> The rouge `}` at the end of the message, after enabling `command={true}`, is now more obvious that is within the source code of the package. I have submitted a PR (https://github.com/skyra-project/discord-components/pull/437) over there to fix this. Might take a bit to get it approved.

This was just merged! Just run pnpm i @skyra/discord-components-react@4.0.0-alpha.26

dpaulos6 committed:
bumping @skyra/discord-components-react to v4.0.0-alpha.26

dpaulos6 commented:
> > The rouge `}` at the end of the message, after enabling `command={true}`, is now more obvious that is within the source code of the package. I have submitted a PR ([skyra-project/discord-components#437](https://github.com/skyra-project/discord-components/pull/437)) over there to fix this. Might take a bit to get it approved.
> 
> This was just merged! Just run pnpm i @skyra/discord-components-react@4.0.0-alpha.26

Done on [65b3866](https://github.com/Would-You-Bot/website-app/pull/41/commits/65b3866ea329639e739e38f1e1ae78f28158a312).
Also when using pnpm its recommended to do `pnpm add` when installing modules.

dpaulos6 reviewed (changes requested):
@DanPlayz0 can you please check the button methods on skyra docs about the emojis? right now we have a custom span with flex in order for it to work, but supposedly `discord-button` has `emoji` and `emojiName` attributes.

If you can take a look at it, i'd really appreciate. If you are unable i can worry about it tomorrow.

DanPlayz0 commented:
> @DanPlayz0 can you please check the button methods on skyra docs about the emojis? right now we have a custom span with flex in order for it to work, but supposedly `discord-button` has `emoji` and `emojiName` attributes.
> 
> If you can take a look at it, i'd really appreciate. If you are unable i can worry about it tomorrow.

I've done a bit of testing with the properties you mentioned.

The `discord-button#emoji` property takes in a url, not an emoji or jsx component. The `discord-button#emojiName` property is just the alternative name for the img element when it doesn't load. It seems we will have to convert all of our hardcoded/inline svgs into files within the public directory.

I am a bit un-experienced with SVGs so when I moved the code for an svg out and into a file, it turned out to look like this.
![image](https://github.com/user-attachments/assets/c062d884-e614-4f33-bbf4-bb16886494b4)
Compared to the original (inline/hardcoded) 
![image](https://github.com/user-attachments/assets/062747a0-9d2e-4b87-b21a-8a1c1290ad6a)
There seems to be some sort of `currentColor` variable set when it's inline.

dpaulos6 commented:
> > @DanPlayz0 can you please check the button methods on skyra docs about the emojis? right now we have a custom span with flex in order for it to work, but supposedly `discord-button` has `emoji` and `emojiName` attributes.
> 
> > 
> 
> > If you can take a look at it, i'd really appreciate. If you are unable i can worry about it tomorrow.
> 
> 
> 
> I've done a bit of testing with the properties you mentioned.
> 
> 
> 
> The `discord-button#emoji` property takes in a url, not an emoji or jsx component. The `discord-button#emojiName` property is just the alternative name for the img element when it doesn't load. It seems we will have to convert all of our hardcoded/inline svgs into files within the public directory.
> 
> 
> 
> I am a bit un-experienced with SVGs so when I moved the code for an svg out and into a file, it turned out to look like this.
> 
> ![image](https://github.com/user-attachments/assets/c062d884-e614-4f33-bbf4-bb16886494b4)
> 
> Compared to the original (inline/hardcoded) 
> 
> ![image](https://github.com/user-attachments/assets/062747a0-9d2e-4b87-b21a-8a1c1290ad6a)
> 
> There seems to be some sort of `currentColor` variable set when it's inline.
> 
> 
> 
> 

i can fix that, push it with the implementation and i'll fix the colors.

DanPlayz0 committed:
Removed double span in discord-button. Using `emoji` and `emojiName` instead

DanPlayz0 commented:
> i can fix that, push it with the implementation and i'll fix the colors.

Just pushed my implementation with commit [672d051](https://github.com/Would-You-Bot/website-app/pull/41/commits/672d051133bcae12d9102d1d6444a7fa0c0614e0)

dpaulos6 committed:
fix external.svg color

dpaulos6 commented:
@DanPlayz0 In this case, the `external.svg` was using `fill="currentColor"`, which is a dynamic property that changes the SVG's vectors to a color set in a CSS class. That allows us to have more accessibility to SVG's within a project, and for example, if you were exporting the SVG as an icon, you could add `className="text-green-500"` to change the SVG's color.

Hope that helps you understand a bit more about SVG's.

DanPlayz0 commented:
> @DanPlayz0 In this case, the `external.svg` was using `fill="currentColor"`, which is a dynamic property that changes the SVG's vectors to a color set in a CSS class. That allows us to have more accessibility to SVG's within a project, and for example, if you were exporting the SVG as an icon, you could add `className="text-green-500"` to change the SVG's color.
> 
> Hope that helps you understand a bit more about SVG's.

I had a feeling it was related to `currentColor`, but I was expecting to have to set a variable to make it work, rather than just changing it to a hex code. Thank for that knowledge!

dpaulos6 reviewed (approved):
This PR is completed in my opinion. Unless there's features in mind to be implemented further, this is ready to be merged.
